### PR TITLE
Update Microsoft.Search to include 2023-11-01 API version

### PR DESCRIPTION
**Pull request by mattgotteiner**
* Add support for 2023-11-01

### CODE REVIEW

File: sdk/search/Azure.ResourceManager.Search/tests/SearchManagementTestBase.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class SearchManagementTestBase : ManagementRecordedTestBase<SearchManagem
     {
         protected ArmClient Client { get; private set; }
 
-        public AzureLocation DefaultLocation => AzureLocation.EastUS;
```
Comment (mattmsft) [inline sdk/search/Azure.ResourceManager.Search/tests/SearchManagementTestBase.cs:17]: @archerzz 
@ArcturusZhang is it possible to re-record tests using the eastus2euap endpoint? Please let me know

**Comment by pallavit** [comment]
@mattmsft this is getting marked as a community contribution. Please review the [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) documentation and ensure that your GitHub organization membership and permissions are correct.
Azure SDK onboarding (Microsoft internal)
[Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) (Microsoft internal)

/cc: @ArthurMa1978 as FYI

**Comment by mattmsft** [comment]
> @mattmsft this is getting marked as a community contribution. Please review the [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) documentation and ensure that your GitHub organization membership and permissions are correct. Azure SDK onboarding (Microsoft internal) [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) (Microsoft internal)
> 
> /cc: @ArthurMa1978 as FYI

I'm a member of the azure-sdk-read team - do I need to be a member of a different team?

**Review by archerzz — COMMENTED** [review]
code changes look fine. need to update test recordings.

**Comment by mattmsft** [comment]
> code changes look fine. need to update test recordings.

I am running into trouble here. Problem:

1. I need to set RESOURCE_MANAGER_URL to eastus2euap.management.azure.com, SERVICE_MANAGEMENT_URL to management.azure.com. We have not fully rolled out service code, it's only available on this ARM endpoint
2. I cannot record tests with these variables set. Hitting an exception in this file sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs
3. if (_recording == null)
            {
                throw new InvalidOperationException("Recorded value should not be set outside the test method invocation");
            }
4. Had a call with @live1206 to discuss this issue. He is looking into it.

I learned we can publish the SDK the same day we merge the PR which is great news. I just want to ensure we have the recordings ready as soon as possible. Please let me know if you know of another way around this problem

**Comment by mattmsft** [comment]
#39383

File: sdk/search/Azure.ResourceManager.Search/src/Generated/Models/QuotaUsageResult.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            Id = id;
+            Unit = unit;
+            CurrentValue = currentValue;
+            Limit = limit;
+            Name = name;
+        }
+
+        /// <summary> The resource id of the quota usage sku endpoint for Microsoft.Search provider. </summary>
+        public string Id { get; }
```
[the remark this answers is not in the document]
Comment (mattmsft) [reply sdk/search/Azure.ResourceManager.Search/src/Generated/Models/QuotaUsageResult.cs]: done

**Comment by mattmsft** [comment]
@jsquire @archerzz not sure if these CI failures are related to test framework issues? please let me know

 Failed CreateOrUpdateTest [2 s]
  Error Message:
   System.InvalidOperationException : Recorded value should not be retrieved outside the test method invocation
  Stack Trace:
 Failed CreateOrUpdateTest [2 s]
  Error Message:
   System.InvalidOperationException : Recorded value should not be retrieved outside the test method invocation
  Stack Trace:

**Comment by live1206** [comment]
@mattmsft You need to Export API after code regeneration.
> Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code. 
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory search'.

And the tests failure was the same cause as https://github.com/Azure/azure-sdk-for-net/issues/39383
Take a look at the [recording](https://github.com/Azure/azure-sdk-assets/blob/5619b84a210258ba7301fdeaff8b55b4eb303971/net/sdk/search/Azure.ResourceManager.Search/tests/SessionRecords/SearchServiceResourceTest/DisableAndEnableLocalAuthAsyncAsync.json#L861), the env var has been saved to the recording. 
And you re-recorded all tests with resource_manager_url, the previous existing tests are failing as well.
So, it seems like we need to fix this bug to unblock this.

**Review by archerzz — APPROVED** [review]
if you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

**Comment by mattmsft** [comment]
> if you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

Updated the change log. Please let me know if you can merge teh PR

**Review by archerzz — APPROVED** [review]

**Pull request merged**